orquesta/conducting.py

```diff
@@ -882,3 +885,40 @@ def get_task_transition_contexts(self, task_id, route):
                 contexts[task_transition_id] = self.get_task_initial_context(t[1], route)
 
         return contexts
+
+    def request_rerun_workflow_status(self, status, rerun_options):
+        # Reset workflow status.
+        current_status = self.get_workflow_status()
+        if current_status != statuses.FAILED:
+            LOG.debug('Workflow execution is not rerunable because its status is not Failed.')
+            return
+
+        self.workflow_state.status = statuses.UNSET
+
```

Comment:
We talked about this. Since this is work done by the conductor, I'm OK with not using the state machines to force this partial reset for rerun. So having said that, you can set the workflow status to `running`.